src/fetchBaseQuery.ts

```diff
@@ -69,13 +139,19 @@ export function fetchBaseQuery({
       ...rest,
     };
 
-    config.headers = prepareHeaders(new Headers(headers), { getState });
+    let tempHeaders = prepareHeaders(new Headers(headers), { getState });
 
-    if (!config.headers.has('content-type')) {
-      config.headers.set('content-type', 'application/json');
+    tempHeaders.forEach((value, key) => {
+      if (value === 'undefined') {
+        tempHeaders.delete(key);
+      }
+    });
+
+    if (!tempHeaders.has('content-type')) {
+      tempHeaders.set('content-type', 'application/json');
     }
 
-    if (body && isPlainObject(body) && isJsonContentType(config.headers)) {
+    if (body && isPlainObject(body) && isJsonContentType(tempHeaders)) {
       config.body = JSON.stringify(body);
     }
 
```

Comment:
None of this really seems like a good idea to me. Instead of passing instances of Headers around, we might want to go the redaxios way and just use objects as it'd be much safer than this. 🤷 I suppose we could also check for `instanceof Headers`, and if false, delete the undefined keys before converting them.

Reply:
how about this?

Reply:
Your approach looks right :). Those dang TS overrides.... learning something new every day here 🎉. I'm going to check the test cases for all of this, and then merge.